pkg/diagnose/runner.go

```diff
@@ -377,61 +344,52 @@ func RunInAgentProcess(diagCfg diagnosis.Config, deps SuitesDepsInAgentProcess)
 	return getDiagnosesFromCurrentProcess(diagCfg, suites)
 }
 
-// RunStdOutInAgentProcess enumerates registered Diagnose suites and get their diagnoses
-// for human consumption
-func RunStdOutInAgentProcess(w io.Writer, diagCfg diagnosis.Config, deps SuitesDepsInAgentProcess) error {
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		return RunInAgentProcess(diagCfg, deps)
-	})
+// RunLocalCheck runs locally the checks created by the registries.
+func RunLocalCheck(diagCfg diagnosis.Config, registries ...func(*diagnosis.Catalog)) ([]diagnosis.Diagnoses, error) {
+	suites := buildCustomSuites(registries...)
+	return getDiagnosesFromCurrentProcess(diagCfg, suites)
 }
 
-// RunStdOutInCLIProcess enumerates registered Diagnose suites and get their diagnoses
-// for human consumption
-func RunStdOutInCLIProcess(w io.Writer, diagCfg diagnosis.Config, deps SuitesDepsInCLIProcess) error {
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		return RunInCLIProcess(diagCfg, deps)
-	})
+// RunDiagnoseStdOut runs the diagnose and outputs the results to the writer.
+func RunDiagnoseStdOut(w io.Writer, diagCfg diagnosis.Config, diagnoses []diagnosis.Diagnoses) error {
+	if diagCfg.JSONOutput {
+		return runStdOutJSON(w, diagnoses)
+	}
+	return runStdOut(w, diagCfg, diagnoses)
 }
 
-// RunStdOutLocalCheck runs locally the checks created by the registries.
-func RunStdOutLocalCheck(w io.Writer, verbose bool, registries ...func(*diagnosis.Catalog)) error {
-	diagCfg := diagnosis.Config{Verbose: verbose, RunLocal: true}
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		suites := buildCustomSuites(registries...)
+func runStdOutJSON(w io.Writer, diagnoses []diagnosis.Diagnoses) error {
+	var count diagnosis.Counters
 
-		return getDiagnosesFromCurrentProcess(diagCfg, suites)
-	})
+	for _, diag := range diagnoses {
+		count.Merge(diag.Counters)
+		diagJSON, err := json.MarshalIndent(diag, "", "  ")
+		if err != nil {
+			body, _ := json.Marshal(map[string]string{"error": fmt.Sprintf("marshalling diagnose results to JSON: %s", err)})
+			fmt.Fprintln(w, string(body))
+			return err
+		}
+		fmt.Fprintln(w, string(diagJSON))
+	}
+	body, _ := json.MarshalIndent(count, "", "  ")
+	fmt.Fprintln(w, string(body))
```

Comment:
I believe this would produce invalid JSON, that won't be able to be parse. At least the JSON example from the PR description is invalid JSON that is not able to be parse.

We would need to ensure all the diagnose are part of a single JSON object including the summary. 

If we want to use Go structs for that I believe we could use something like:

We would need a new struct which stores all the diagnoses 


```go
type DiagnosisResult struct {
	Diagnoses []Diagnoses `json:"diagnoses"`
	Summary   Counters    `json:"summary"`
}
```

We would use this new struct and Marshall it as a single unit

```go
func runStdOutJSON(w io.Writer, diagnoses []diagnosis.Diagnoses) error {
	var summaryCounters diagnosis.Counters
	result := diagnosis.DiagnosisResult{
		Diagnoses: diagnoses,
	}

	for _, diag := range diagnoses {
		summaryCounters.Merge(diag.Counters)
	}

	result.Summary = summaryCounters

	diagJSON, err := json.MarshalIndent(result, "", "  ")
	if err != nil {
		body, _ := json.Marshal(map[string]string{"error": fmt.Sprintf("marshalling diagnose results to JSON: %s", err)})
		fmt.Fprintln(w, string(body))
		return err
	}
	fmt.Fprintln(w, string(diagJSON))

	return nil
}
```

Reply:
Yes, did not wanted to create a new struct only for the JSON one, if we create one I will refactor the code to remove the `Counters` in
```
// Diagnoses is a collection of Diagnosis
type Diagnoses struct {
	SuiteName      string      `json:"suite_name"`
	SuiteDiagnoses []Diagnosis `json:"diagnosis"`
	Counters       Counters    `json:"counters"`
}
```
so we only compute it at one time in `DiagnosisResult` so we stick to the diagnose -v avoiding having multiple Counters in the JSON output, does it seems right for you ?